packages/store/addon/-private/system/snapshot.ts

```diff
@@ -362,18 +414,20 @@ export default class Snapshot {
         let internalModel = store._internalModelForResource(member);
         if (!internalModel.isDeleted()) {
           if (ids) {
-            results.push(member.id);
+            (results as RecordId[]).push(member.id);
           } else {
-            results.push(internalModel.createSnapshot());
+            (results as Snapshot[]).push(internalModel.createSnapshot());
           }
```

Comment:
Same request as above for a comment explaining what makes this hold.

Reply:
I ended up with a slightly different approach due to how this is threaded through the whole method for both `belongsTo` and `hasMany`, basically these methods are used in one of two "modes", returning id/s or returning snapshot/s. I renamed `id` and `ids` to `returnModeIsId` and `returnModeIsIds` to clarify this.